src/app/main.cpp

```diff
@@ -67,17 +84,18 @@ static prometheus::Exposer create_exposer(Context& context) {
 }
 static Context make_context() {
   auto config = shar::Config::parse_from_file("config.json");
-  auto logger  = Logger("server.log");
+  auto logger = Logger("server.log");
   auto metrics = std::make_shared<shar::Metrics>(logger);
 
-  return Context{config, logger, metrics};
+  return Context{ config, logger, metrics };
 }
 
 static int run() {
   auto context = make_context();
   auto exposer = create_exposer(context);
-  auto capture = create_capture(context);
-  auto encoder = create_encoder(context);
+  const auto& monitor = select_monitor(context);
```

Comment:
There is no reason to rely on "const reference lifetime extension" feature in C++17.